ably/realtime/connection.py

```diff
@@ -49,12 +52,20 @@ async def connect(self):
 
     async def close(self):
         self.__state = ConnectionState.CLOSING
-        if self.__websocket:
-            await self.__websocket.close()
+        self.__closed_future = asyncio.Future()
+        if self.__websocket and self.__state == ConnectionState.CONNECTED:
+            task = asyncio.create_task(self.close_connection())
+            await task
```

Comment:
A couple of things here:
- creating a task from a coroutine with `create_task` and then immediately awaiting it is the same as just awaiting the coroutine, so these two lines could be replaced with `await self.close_connection()`.
- `self.__closed_future` isn't actually being used anywhere. I think it should be awaited as part of this if statement because otherwise the `close` coroutine will finish before `CLOSED` message is received.